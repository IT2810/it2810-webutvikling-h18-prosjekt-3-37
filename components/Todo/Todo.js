import React from 'react';
import { StyleSheet, View } from 'react-native';

import { Body, Text, ListItem, CheckBox } from 'native-base';

export default class Todo extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: (this.props.name === null) ? "Default name" : this.props.name,
    };
  }

  render() {
    return (
      <ListItem>
        <CheckBox onPress={() => this.props.onClick()} checked={ this.props.checked } />
        <Body><Text> { this.state.name } </Text></Body>
      </ListItem>
    )
  }
}
