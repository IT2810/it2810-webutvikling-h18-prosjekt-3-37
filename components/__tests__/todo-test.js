import React from 'react'
import renderer from 'react-test-renderer'
import Todo from '../Todo/Todo'

//Check that AddTodo-Component renders correctly and matches snapshot
it('renders correctly', () => {
    const tree = renderer.create(<Todo />).toJSON()
    expect(tree).toMatchSnapshot()
})


