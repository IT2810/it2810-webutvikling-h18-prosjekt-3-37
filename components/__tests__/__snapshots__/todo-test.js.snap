// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderBottomWidth": 0.5,
        "borderColor": "#c9c9c9",
        "flexDirection": "row",
        "marginLeft": 16,
        "paddingBottom": 13,
        "paddingRight": 16,
        "paddingTop": 13,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#039BE5",
          "borderLeftWidth": 1,
          "borderRadius": 13,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "height": 20,
          "left": 10,
          "marginLeft": -10,
          "marginRight": 10,
          "opacity": 1,
          "overflow": "hidden",
          "paddingBottom": 0,
          "paddingLeft": 3,
          "width": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": null,
          "alignSelf": null,
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
            "marginLeft": 10,
            "marginRight": 10,
          }
        }
        uppercase={false}
      >
         
         
      </Text>
    </View>
  </View>
</View>
`;
